File: CS/HowToAddCustomControl/CustomLabel.cs
using System.ComponentModel;
using DevExpress.XtraReports.UI;
using DevExpress.Utils.Serializing;

namespace Sample  {   
    public class CustomLabel : XRLabel
    {
        private string _someProperty;
        private string _bindableProperty;

        [Browsable(true), Bindable(false), Category("Data")]
        [XtraSerializableProperty]
        [DefaultValue("")]
        [DesignerSerializationVisibility(DesignerSerializationVisibility.Visible), EditorBrowsable(EditorBrowsableState.Always)]
        public string SomeProperty
        {
            get { return this._someProperty; }
            set { this._someProperty = value; }
        }

        [Browsable(true), Bindable(false), Category("Data")]
        [XtraSerializableProperty]
        [DefaultValue("")]
        [DesignerSerializationVisibility(DesignerSerializationVisibility.Visible), EditorBrowsable(EditorBrowsableState.Always)]
        public string BindableProperty
        {
            get { return this._bindableProperty; }
            set { this._bindableProperty = value; }
        }
    }
}

File: VB/HowToAddCustomControl/CustomLabel.vb
Imports System.ComponentModel
Imports DevExpress.XtraReports.UI
Imports DevExpress.Utils.Serializing

Namespace Sample

    Public Class CustomLabel
        Inherits XRLabel

        Private _someProperty As String

        Private _bindableProperty As String

        <Browsable(True), Bindable(False), Category("Data")>
        <XtraSerializableProperty>
        <DefaultValue("")>
        <DesignerSerializationVisibility(DesignerSerializationVisibility.Visible), EditorBrowsable(EditorBrowsableState.Always)>
        Public Property SomeProperty As String
            Get
                Return _someProperty
            End Get

            Set(ByVal value As String)
                _someProperty = value
            End Set
        End Property

        <Browsable(True), Bindable(True), Category("Data")>
        <XtraSerializableProperty>
        <DefaultValue("")>
        <DesignerSerializationVisibility(DesignerSerializationVisibility.Visible), EditorBrowsable(EditorBrowsableState.Always)>
        Public Property BindableProperty As String
            Get
                Return _bindableProperty
            End Get

            Set(ByVal value As String)
                _bindableProperty = value
            End Set
        End Property

        Protected Overrides Sub PutStateToBrick(ByVal brick As DevExpress.XtraPrinting.VisualBrick, ByVal ps As DevExpress.XtraPrinting.PrintingSystemBase)
            MyBase.PutStateToBrick(brick, ps)
            brick.Text = BindableProperty
        End Sub
    End Class
End Namespace
